core/rpc/client/jsonrpc.go

```diff
@@ -0,0 +1,214 @@
+// Package rpcclient provides a Kwil JSON-RPC (API v1) client. It supports only
+// HTTP POST, no WebSockets yet. The JSONRPCClient type implements the
+// core/rpc/client/user.TxSvcClient interface that is required by
+// core/client.JSONRPCClient.
+package client
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"reflect"
+	"strconv"
+	"sync/atomic"
+
+	"github.com/kwilteam/kwil-db/core/log"
+	jsonrpc "github.com/kwilteam/kwil-db/core/rpc/json"
+)
+
+// JSONRPCClient will use the commands to make certain requests
+//  - "params" field is set to the marshalled request structs
+//  - the "method" in the outer request type is instead of the endpoint, all POST
+//  - "id" is set to a counter's value
+//	- "jsonrpc" set to "2.0"
+//	- make http POST request with marshalled Request
+//	- unmarshal response into Response
+//	- if "error" field is set, the Error type is returned as Go error
+//	- maybe compare "id" to expected ID from request
+//	- unmarshal the "result" (json.RawMessage) into the method's result type
+
+// note compared to http grpc gateway:
+//
+//  - the structs marshal into json objects, set in "params" field of request.
+//    with the old http gateway, endpoints were defined with associated
+//    message (pb) types sent in POST, or just GET with endpoint implying method.
+//  - the "method" in the outer request type is instead of the endpoint, all POST
+
+// JSONRPCClient is a JSON-RPC client that handles sending and receiving
+// JSON-RPC requests and responses. It is a low-level client that does not
+// care about the specifics of the JSON-RPC methods or responses.
+type JSONRPCClient struct {
```

Comment:
Well the naming is kind awkward, I tempted to name it like `BaseRPCClient`.  The reason is in the future say we support WS, we might have another base client

Reply:
Meh.  But if we support websocket, that's still json-rpc, just using websocket.  Anyway, I think `JSONRPCClient` is appropriate, at least while it is defined in `core/rpc/client`.